Unable to build sfm module

- OpenCV => 4.0.0 opencv and opencv_contrib even mwith maste
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017

I deleted my opencv build repo and rebuild opencv and opencv_contrib and run cmake
[CMakeCache.txt](https://github.com/opencv/opencv_contrib/files/2544950/CMakeCache.txt)

I have got errors building opencv_contrib : 

```
1>------ Build started: Project: opencv_sfm, Configuration: Debug x64 ------
1>opencv_sfm_pch.cpp
1>g:\lib\opencv_contrib\modules\sfm\include\opencv2\sfm\simple_pipeline.hpp(87): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\include\opencv2\sfm\simple_pipeline.hpp(88): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
1>conditioning.cpp
1>fundamental.cpp
1>io.cpp
1>numeric.cpp
1>projection.cpp
1>reconstruct.cpp
1>robust.cpp
1>simple_pipeline.cpp
1>triangulation.cpp
1>opencv_sfm_main.cpp
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(63): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(63): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(63): error C2988: unrecognizable template declaration/definition (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(63): error C2143: syntax error: missing ',' before '<' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(81): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(81): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(81): error C2988: unrecognizable template declaration/definition (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(81): error C2143: syntax error: missing ',' before '<' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(92): error C2995: 'void cv::eigen2cv(const int)': function template has already been defined (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(63): note: see declaration of 'cv::eigen2cv' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(96): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(96): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(124): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(124): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(142): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(142): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(170): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(170): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(189): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(189): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(209): error C2995: 'void cv::cv2eigen(const cv::Mat &)': function template has already been defined (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(141): note: see declaration of 'cv::cv2eigen' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(214): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(214): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(235): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(235): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(259): error C2653: 'Eigen': is not a class or namespace name (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv\modules\core\include\opencv2\core\eigen.hpp(259): error C2061: syntax error: identifier 'Matrix' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(75): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(76): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(77): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(78): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(81): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(82): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(83): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\io.cpp(87): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(100): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(100): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(136): warning C4456: declaration of 'nviews' hides previous local declaration
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(100): note: see declaration of 'nviews'
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(136): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(136): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(143): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(167): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(167): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(211): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(211): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\reconstruct.cpp(218): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\matches.h(120): warning C4267: '=': conversion from 'size_t' to 'libmv::Matches::ImageID', possible loss of data (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\matches.h(127): warning C4267: '=': conversion from 'size_t' to 'libmv::Matches::TrackID', possible loss of data (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\triangulation.cpp(155): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\triangulation.cpp(181): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\feature_matching.h(42): warning C4458: declaration of 'descriptor' hides class member (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\feature_matching.h(55): note: see declaration of 'KeypointFeature::descriptor' (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\simple_pipeline.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\base\vector.h(79): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\robust.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\base\vector.h(79): note: while compiling class template member function 'int libmv::vector<int,Eigen::aligned_allocator<T>>::size(void) const'
1>        with
1>        [
1>            T=int
1>        ] (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\robust.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\robust.cpp(79): note: see reference to function template instantiation 'int libmv::vector<int,Eigen::aligned_allocator<T>>::size(void) const' being compiled
1>        with
1>        [
1>            T=int
1>        ]
1>g:\lib\opencv_contrib\modules\sfm\src\robust.cpp(68): note: see reference to class template instantiation 'libmv::vector<int,Eigen::aligned_allocator<T>>' being compiled
1>        with
1>        [
1>            T=int
1>        ]
1>g:\lib\opencv_contrib\modules\sfm\src\robust.cpp(114): note: see reference to function template instantiation 'double cv::sfm::fundamentalFromCorrespondences8PointRobust<float>(const cv::Mat_<float> &,const cv::Mat_<float> &,const double,cv::Mat_<float>,std::vector<int,std::allocator<_Ty>> &,const double)' being compiled
1>        with
1>        [
1>            _Ty=int
1>        ]
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\base\vector.h(116): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\robust.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\base\vector.h(115): note: while compiling class template member function 'void libmv::vector<int,Eigen::aligned_allocator<T>>::clear(void)'
1>        with
1>        [
1>            T=int
1>        ] (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\robust.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\base\vector.h(49): note: see reference to function template instantiation 'void libmv::vector<int,Eigen::aligned_allocator<T>>::clear(void)' being compiled
1>        with
1>        [
1>            T=int
1>        ] (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\robust.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\fundamental.cpp(255): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
1>g:\lib\opencv_contrib\modules\sfm\src\fundamental.cpp(294): note: see reference to function template instantiation 'void cv::sfm::motionFromEssential<float>(const cv::Mat_<float> &,std::vector<cv::Mat,std::allocator<_Ty>> &,std::vector<_Ty,std::allocator<_Ty>> &)' being compiled
1>        with
1>        [
1>            _Ty=cv::Mat
1>        ]
1>g:\lib\opencv_contrib\modules\sfm\src\fundamental.cpp(549): warning C4100: 's': unreferenced formal parameter
1>g:\lib\opencv_contrib\modules\sfm\src\fundamental.cpp(586): note: see reference to function template instantiation 'void cv::sfm::computeOrientation<float>(const cv::Mat_<float> &,const cv::Mat_<float> &,cv::Mat_<float>,cv::Mat_<float>,T)' being compiled
1>        with
1>        [
1>            T=float
1>        ]
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\numeric\numeric.h(39): warning C4505: 'sincos': unreferenced local function has been removed (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\numeric.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\numeric\numeric.h(39): warning C4505: 'sincos': unreferenced local function has been removed (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\robust.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\numeric\numeric.h(39): warning C4505: 'sincos': unreferenced local function has been removed (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\triangulation.cpp)
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\numeric\numeric.h(39): warning C4505: 'sincos': unreferenced local function has been removed (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\conditioning.cpp)
1>g:\lib\eigen\eigen\src\core\util\memory.h(278): warning C4702: unreachable code
1>g:\lib\eigen\eigen\src\core\util\memory.h(275): warning C4701: potentially uninitialized local variable 'i' used
1>g:\lib\eigen\eigen\src\core\util\memory.h(278): warning C4702: unreachable code
1>g:\lib\eigen\eigen\src\core\util\memory.h(275): warning C4701: potentially uninitialized local variable 'i' used
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\numeric\numeric.h(39): warning C4505: 'sincos': unreferenced local function has been removed (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\fundamental.cpp)
1>g:\lib\eigen\eigen\src\core\util\memory.h(278): warning C4702: unreachable code
1>g:\lib\eigen\eigen\src\core\util\memory.h(275): warning C4701: potentially uninitialized local variable 'i' used
1>g:\lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\numeric\numeric.h(39): warning C4505: 'sincos': unreferenced local function has been removed (compiling source file G:\Lib\opencv_contrib\modules\sfm\src\projection.cpp)
1>Done building project "opencv_sfm.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

```

If I forget [commit ](https://github.com/opencv/opencv_contrib/commit/fc28d02a1e4a2547c9f3d169556863f1fffa539a#diff-c9222eba8f7be872ebc0898a3ea2cee7)compilation ad link is OK

--- a/modules/sfm/src/simple_pipeline.cpp
+++ b/modules/sfm/src/simple_pipeline.cpp
@@ -37,6 +37,9 @@
 
 #if CERES_FOUND
 
+// Eigen
+#include <Eigen/Core>
+
 #include <opencv2/core/eigen.hpp>
 #include <opencv2/xfeatures2d.hpp>
 

